src/data/real/liouville.lean

```diff
@@ -56,6 +58,69 @@ begin
   exact not_le.mpr a1 (nat.mul_lt_mul_pow_succ (int.coe_nat_pos.mp ap) (int.coe_nat_lt.mp q1)).le,
 end
 
-end is_liouville
+end liouville
+
+open polynomial metric set real ring_hom
+
+/-- This lemma collects the properties needed to prove `exists_pos_real_of_irrational_root`.
+It is stated in more general form than needed: in the intended application, `Z = ℤ`, `N = ℕ`,
+`R = ℝ`, `d a = a ^ f.nat_degree`, `j z a  = z / (a + 1)`, `f ∈ ℤ[x]`, `α` is an irrational
+root of `f`, `ε` is small, `M` is a bound on the Lipschitz constant of `f` near `α`, `n` is
+the degree of the polynomial `f`.
+-/
+lemma exists_one_le_pow_mul_dist {Z N R : Type*} [metric_space R]
+  {d : N → ℝ} {j : Z → N → R} {f : R → R} {α : R} {ε M : ℝ}
+--denominators are positive
+  (d0 : ∀ (a : N), 1 ≤ d a)
+  (e0 : 0 < ε)
+--function is Lipschitz at α
+  (B : ∀ ⦃y : R⦄, y ∈ closed_ball α ε → dist (f α) (f y) ≤ (dist α y) * M)
+--clear denominators
+  (L : ∀ ⦃z : Z⦄, ∀ ⦃a : N⦄, j z a ∈ closed_ball α ε → 1 ≤ (d a) * dist (f α) (f (j z a))) :
+  ∃ e : ℝ, 0 < e ∧ ∀ (z : Z), ∀ (a : N), 1 ≤ (d a) * (dist α (j z a) * e) :=
+begin
+  have me0 : 0 < max (1 / ε) M := lt_max_iff.mpr (or.inl (one_div_pos.mpr e0)),
+  refine ⟨max (1 / ε) M, me0, λ z a, _⟩,
+  by_cases dm1 : 1 ≤ (dist α (j z a) * max (1 / ε) M),
+  { exact one_le_mul_of_one_le_of_one_le (d0 a) dm1 },
+  { have : j z a ∈ closed_ball α ε,
+    { refine mem_closed_ball'.mp (le_trans _ ((one_div_le me0 e0).mpr (le_max_left _ _))),
+      exact ((le_div_iff me0).mpr (not_le.mp dm1).le) },
+    refine (L this).trans _,
+    refine mul_le_mul_of_nonneg_left ((B this).trans _) (zero_le_one.trans (d0 a)),
+    exact mul_le_mul_of_nonneg_left (le_max_right _ M) dist_nonneg }
+end
+
+lemma exists_pos_real_of_irrational_root {α : ℝ} (ha : _root_.irrational α)
+  {f : polynomial ℤ} (f0 : f ≠ 0) (fa : eval α (map (algebra_map ℤ ℝ) f) = 0):
+  ∃ ε : ℝ, 0 < ε ∧
+    ∀ (a : ℤ), ∀ (b : ℕ), (1 : ℝ) ≤ (b.succ) ^ f.nat_degree * (abs (α - (a / (b.succ))) * ε) :=
+begin
+  set fR : polynomial ℝ := map (algebra_map ℤ ℝ) f,
+  obtain fR0 : fR ≠ 0 := λ fR0, (map_injective (algebra_map ℤ ℝ) (λ _ _ A, int.cast_inj.mp A)).ne
+    f0 (fR0.trans (polynomial.map_zero _).symm),
+  have ar : α ∈ (fR.roots.to_finset : set ℝ) :=
+    finset.mem_coe.mpr (multiset.mem_to_finset.mpr ((mem_roots fR0).mpr (is_root.def.mpr fa))),
+  obtain ⟨ζ, z0, U⟩ :=
+    @exists_closed_ball_inter_eq_singleton_of_discrete _ _ _ discrete_of_t1_of_finite _ ar,
+  obtain ⟨xm, ⟨h_x_max_range, hM⟩⟩ := is_compact.exists_forall_ge (@compact_Icc (α - ζ) (α + ζ))
```

Comment:
```suggestion
  obtain ⟨xm, h_x_max_range, hM⟩ := is_compact.exists_forall_ge (@compact_Icc (α - ζ) (α + ζ))
```
is the same thing here

Reply:
I am going to wait for CI to finish, just to confirm that everything builds and I will commit this suggestion once that happens!